python/cuspatial/cuspatial/tests/binpreds/test_intersects_only_binpreds.py

```diff
@@ -0,0 +1,982 @@
+    linestring_generator, multilinestring_generator
+):
+    g1 = cuspatial.GeoSeries([*linestring_generator(100, 5)])
+    g2 = cuspatial.GeoSeries([*multilinestring_generator(100, 5, 5)])
+    gpdg1 = g1.to_geopandas()
+    gpdg2 = g2.to_geopandas()
+    got = g1.intersects(g2)
+    expected = gpdg1.intersects(gpdg2)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_100_multilinestrings_intersects_100_linestrings(
+    linestring_generator, multilinestring_generator
+):
+    g1 = cuspatial.GeoSeries([*multilinestring_generator(100, 5, 5)])
+    g2 = cuspatial.GeoSeries([*linestring_generator(100, 5)])
+    gpdg1 = g1.to_geopandas()
+    gpdg2 = g2.to_geopandas()
+    got = g1.intersects(g2)
+    expected = gpdg1.intersects(gpdg2)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_100_multilinestrings_intersects_100_multilinestrings(
+    multilinestring_generator,
+):
+    g1 = cuspatial.GeoSeries([*multilinestring_generator(100, 5, 5)])
+    g2 = cuspatial.GeoSeries([*multilinestring_generator(100, 5, 5)])
+    gpdg1 = g1.to_geopandas()
+    gpdg2 = g2.to_geopandas()
+    got = g1.intersects(g2)
+    expected = gpdg1.intersects(gpdg2)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_linestring_intersects_multilinestring():
+    g1 = cuspatial.GeoSeries(
+        [
+            LineString([(0.0, 0.0), (0.0, 1.0)]),
+            LineString([(0.0, 1.0), (1.0, 1.0)]),
+        ]
+    )
+    g2 = cuspatial.GeoSeries(
+        [
+            MultiLineString(
+                [
+                    [(0.0, 0.0), (0.0, 1.0)],
+                    [(0.0, 1.0), (1.0, 1.0)],
+                ]
+            ),
+            MultiLineString(
+                [
+                    [(0.0, 0.0), (0.0, 1.0)],
+                    [(0.0, 1.0), (1.0, 1.0)],
+                ]
+            ),
+        ]
+    )
+    gpdg1 = g1.to_geopandas()
+    gpdg2 = g2.to_geopandas()
+    got = g1.intersects(g2)
+    expected = gpdg1.intersects(gpdg2)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_multilinestring_intersects_linestring():
+    g1 = cuspatial.GeoSeries(
+        [
+            MultiLineString(
+                [
+                    [(0.0, 0.0), (0.0, 1.0)],
+                    [(0.0, 1.0), (1.0, 1.0)],
+                ]
+            ),
+            MultiLineString(
+                [
+                    [(0.0, 0.0), (0.0, 1.0)],
+                    [(0.0, 1.0), (1.0, 1.0)],
+                ]
+            ),
+        ]
+    )
+    g2 = cuspatial.GeoSeries(
+        [
+            LineString([(0.0, 0.0), (0.0, 1.0)]),
+            LineString([(0.0, 1.0), (1.0, 1.0)]),
+        ]
+    )
+    gpdg1 = g1.to_geopandas()
+    gpdg2 = g2.to_geopandas()
+    got = g1.intersects(g2)
+    expected = gpdg1.intersects(gpdg2)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_linestring_intersects_polygon():
```

Comment:
Indeed, there are many special cases of polygons that will need to have tests written for them. I'm planning on writing a suite of special case tests when the dispatched tests, in the next upcoming PR for 23.06, are complete. All of the special case tests will be documented and need pretty careful scrutiny.

How do you feel about postponing them until then so that this can get into 23.04? Exhaustively covering special cases is going to be a multi-day effort I think.

Reply:
> How do you feel about postponing them until then so that this can get into 23.04?

I'm totally fine with postponing as long as they are tracked. You can convert this comment into an issue, for example.

Reply:
Thank you, and [done](https://github.com/rapidsai/cuspatial/issues/1036)